packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,85 +503,399 @@ export const Tabs = (props: TabsProps) => {
+        {
+          'iui-green': color === 'green',
+          'iui-animated': type !== 'default' && isClient,
+          'iui-not-animated': type !== 'default' && !isClient,
+          'iui-large': hasSublabel,
+        },
+        className,
+      )}
+      data-iui-overflow={overflowOptions?.useOverflow}
+      data-iui-scroll-placement={scrollingPlacement}
+      role='tablist'
+      onKeyDown={onKeyDown}
+      ref={refs}
+      {...rest}
+    >
+      {items.map((item, index) => {
+        return (
+          <li key={index}>
+            <TabsContext.Provider
+              value={{
+                index,
+                currentActiveIndex,
+                setCurrentActiveIndex,
+                onTabSelected: onTabClick,
+                setFocusedIndex,
+                hasSublabel,
+                setHasSublabel,
+              }}
+              key={index}
+            >
+              {item}
+            </TabsContext.Provider>
+          </li>
+        );
+      })}
+    </Box>
   );
+}) as PolymorphicForwardRefComponent<'ul', TabsTabListOwnProps>;
+TabsTabList.displayName = 'Tabs.TabList';
+
+// ----------------------------------------------------------------------------
+// Tabs.Tab component
+
+type TabsTabOwnProps = {
+  /**
+   * Tab label used for simple Tab construction.
+   * Cannot be used with tabs that have icons or descriptions.
+   */
+  label?: string | React.ReactNode;
+};
+
+const TabsTab = React.forwardRef((props, ref) => {
+  const { className, children, label, ...rest } = props;
 
+  const { index, currentActiveIndex, onTabSelected, setFocusedIndex } =
+    useSafeContext(TabsContext);
+
+  const onClick = () => {
+    if (index !== undefined) {
+      setFocusedIndex && setFocusedIndex(index);
+      onTabSelected && onTabSelected(index);
+    }
+  };
   return (
     <Box
-      className={cx('iui-tabs-wrapper', `iui-${orientation}`, wrapperClassName)}
-      style={stripeProperties}
+      as='button'
+      className={cx(
+        'iui-tab',
+        { 'iui-active': index === currentActiveIndex },
+        className,
+      )}
+      role='tab'
+      tabIndex={index === currentActiveIndex ? 0 : -1}
+      onClick={onClick}
+      aria-selected={index === currentActiveIndex}
+      ref={ref}
+      {...rest}
     >
+      {label ? (
+        <Tabs.TabInfo>
+          <Tabs.TabLabel>{label}</Tabs.TabLabel>
+        </Tabs.TabInfo>
+      ) : (
+        children
+      )}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'button', TabsTabOwnProps>;
+TabsTab.displayName = 'Tabs.Tab';
+
+// ----------------------------------------------------------------------------
+// Tabs.TabIcon component
+
+const TabsTabIcon = polymorphic.span('iui-tab-icon', {
+  'aria-hidden': true,
+});
+TabsTabIcon.displayName = 'Tabs.TabIcon';
+
+// ----------------------------------------------------------------------------
+// Tabs.TabInfo component
+
+const TabsTabInfo = polymorphic.span('iui-tab-label');
```

Comment:
since we have the actual label be under `TabInfo` now, maybe this should be named to `iui-tab-info` or similar, and the actual label will have `iui-tab-label` as class name?

Reply:
`iui-tab-label` has changed to `iui-tab-info`

I did not add the `iui-tab-label` class to `Tabs.TabLabel` since there are no selectors which require that element's class name. CSS attributes are passed using the `> *` selector from the `iui-tab-info` mixin

Reply:
Due to the changes made in the DOM structure, `TabInfo` has been removed